miner/miner.go

```diff
@@ -303,6 +303,7 @@ func (m *Miner) GetBestMiningCandidate(ctx context.Context) (*MiningBase, error)
 		}
 		ltsw, err := m.api.ChainTipSetWeight(ctx, m.lastWork.TipSet.Key())
 		if err != nil {
+			m.lastWork = nil
```

Comment:
Whats the actual bug being fixed here? Shouldnt we also have the same line copied a few lines up?

Reply:
lotus/miner/miner.go
```golang
func (m *Miner) mine(ctx context.Context) {
	...
	for {
		prebase, err := m.GetBestMiningCandidate(ctx)
		// If an error occurs in GetBestMiningCandidate, it will wait 5 seconds to retry GetBestMiningCandidate
		if err != nil {
			log.Errorf("failed to get best mining candidate: %s", err)
			m.niceSleep(time.Second * 5)
			continue
		}
		...
	}
	...
}

func (m *Miner) GetBestMiningCandidate(ctx context.Context) (*MiningBase, error) {
	...
	if m.lastWork != nil {
		if m.lastWork.TipSet.Equals(bts) {
			return m.lastWork, nil
		}

		btsw, err := m.api.ChainTipSetWeight(ctx, bts.Key())
		if err != nil {
			return nil, err
		}
		ltsw, err := m.api.ChainTipSetWeight(ctx, m.lastWork.TipSet.Key())
		// If lastWork is using an incorrect TipSet, an error is received and returned here, 
		// for example i have got: blockstore: block not found. 
		// There is no modification lastWork here, so if you try it again, you will get the same error here. 
		// In the mine() function above, it tries again and again, but always got this error
		if err != nil {
			return nil, err
		}
		...
	}
	...
}
```

Reply:
Ah! Thank you for that. This makes sense